packages/nimble-components/src/table/specs/table-column-specs/table-column-anchor-hld.md

```diff
@@ -0,0 +1,158 @@
+
+The first part of this task is to detect which click events have come from a `nimble-anchor`. We can't just look at the event's `target`, because that is changed to the host control upon crossing a shadow DOM boundary. We can, however, look at the `composedPath()` of the event, which will include every element that was along the bubble-up path. If we find a `nimble-anchor` among them (by using `instanceof`), we know we should handle the event.
+
+To handle the event, we must find the associated anchor column directive. We have the `nimble-anchor` that the event came from, and by calling `anchor.getRootNode().host`, we can get the `nimble-table-cell`. Currently, there is no way to get from a given table cell to its associated column. One way to solve this is to include the associated `columnId` in the cell state, i.e. `cell.cellState.columnId`.
+
+```ts
+export interface TableCellState<...> {
+    cellRecord: TCellRecord;
+    columnConfig: TColumnConfig;
+    columnId: string | undefined;
+}
+```
+
+`columnId` is at the top level rather than part of the `columnConfig`, because it applies for all column types, and it is not for configuration.
+Adding `columnId` to the cell state would be a change to `nimble-components` just to support `nimble-angular`, but it seems generic enough to be a reasonable API change.
+
+One problem is that `columnId` is `undefined` unless a client sets it. We would also need to start auto-generating unique `columnId` values if the client hasn't provided one (i.e. in `connectedCallback`). An alternative would be to document that users must provide column ids when wishing to integrate with the Angular router, but this seems like an arbitrary requirement that would be better to avoid. Another option considered was to reuse the existing `internalUniqueId` that each column defines, but this is used for the action menu slot, and overloading its use as a general-purpose ID seems wrong.
+
+Another issue is that the client may change the `columnId` programmatically, invalidating the value stored in the cell state. To ensure consistency, we must have a `columnIdChanged()` handler that causes `refreshRows()` to be called on the table.
+
+Now that we have the `columnId` for a given anchor click, we need to get the column directive. We can get all potential directives as content children:
+
+```ts
+@ContentChildren(NimbleTableColumnAnchorDirective) public anchorColumnDirectives: QueryList<NimbleTableColumnAnchorDirective>;
+```
+
+Each directive has a reference to its associated element, so we can get to the `nimble-table-column-anchor`. From there, we have access to the `columnId`, so we can find the match.
+
+Finally, we need to call `RouterLinkWithHref.onClick()` to do the router navigation. However, we'll first need to set up the `routerLink` property with the url:
+
+```ts
+public doRouterNavigation(href: string): boolean {
```

Comment:
What's going to track the event information such as the mouse button that was clicked and whether or not ctrl and/or shift were pressed? Is that going to be part of the information passed to `doRouterNavigation()`?

Reply:
I've added a note about that. Basically, the RouterLink code only handles plain left-clicks, so we'll check the event state and only call `doRouterNavigation()` when `button===0` and all key flags are false.